mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp

```diff
@@ -291,8 +301,9 @@ LogicalResult backwardWeightAtomicAdd(Conv2DBwdWeightOp op,
     addKBlockWrap.passThrough({"k", "c", "y", "x"});
 
     TransformMapAttr addKBlockTransformAttr = addKBlockTransform.get();
-    Value withKBlock =
-        b.create<TransformOp>(loc, op.filter(), addKBlockTransformAttr);
+    Value filterTensorInUse = (hasWorkspace) ? op.workspace()[0] : op.filter();
+    Value withKBlock = b.create<miopen::TransformOp>(loc, filterTensorInUse,
+                                                     addKBlockTransformAttr);
```

Comment:
Out of curiosity, is this the key logic for using the workspace? I'm asking because I don't see much relevant logic below gridwise lowering.

Reply:
Yes. This is where we specify the true output in a backward weight convolution when we use atomic add.